streams/src/main/java/org/apache/kafka/streams/state/StateSerdes.java

```diff
@@ -165,7 +166,18 @@ public V valueFrom(byte[] rawValue) {
      * @return     the serialized key
      */
     public byte[] rawKey(K key) {
-        return keySerde.serializer().serialize(topic, key);
+        try {
+            return keySerde.serializer().serialize(topic, key);
+        } catch (final ClassCastException e) {
+            final String keyClass = key == null ? "unknown because key is null" : key.getClass().getName();
```

Comment:
If key is null, would `serialize` really throw the `ClassCastException`? I think in this case it would throw NPE right?